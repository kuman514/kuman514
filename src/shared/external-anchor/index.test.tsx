import { render } from '@testing-library/react';

import ExternalAnchor from '.';

describe('ExternalAnchor', () => {
  it('should show as expected', () => {
    const { container } = render(
      <ExternalAnchor href="https://yasuomaverick.kuman514.xyz/">
        야스오매버릭 홈페이지로 가기
      </ExternalAnchor>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should have a link that equals the href & _blank as the default target', () => {
    const { container } = render(
      <ExternalAnchor href="https://yasuomaverick.kuman514.xyz/">
        야스오매버릭 홈페이지로 가기
      </ExternalAnchor>
    );
    expect((container.firstChild as HTMLAnchorElement).href).toStrictEqual(
      'https://yasuomaverick.kuman514.xyz/'
    );
    expect((container.firstChild as HTMLAnchorElement).target).toStrictEqual(
      '_blank'
    );
  });

  it('should have custom target & custom class name', () => {
    const { container } = render(
      <ExternalAnchor
        href="https://yasuomaverick.kuman514.xyz/"
        customTarget="_parent"
        customClassName="text-2xl underline"
      >
        야스오매버릭 홈페이지로 가기
      </ExternalAnchor>
    );
    expect(container.firstChild).toMatchSnapshot();
    expect((container.firstChild as HTMLAnchorElement).target).toStrictEqual(
      '_parent'
    );
  });
});
