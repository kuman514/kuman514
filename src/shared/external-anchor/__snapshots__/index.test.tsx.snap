// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalAnchor should have custom target & custom class name 1`] = `
<a
  class="text-blue-500 hover:text-purple-500 text-2xl underline"
  href="https://yasuomaverick.kuman514.xyz/"
  target="_parent"
>
  야스오매버릭 홈페이지로 가기
</a>
`;

exports[`ExternalAnchor should show as expected 1`] = `
<a
  class="text-blue-500 hover:text-purple-500 undefined"
  href="https://yasuomaverick.kuman514.xyz/"
  target="_blank"
>
  야스오매버릭 홈페이지로 가기
</a>
`;
